optuna/artifacts/_download.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from optuna._experimental import experimental_func
+from optuna.artifacts._protocol import ArtifactStore
+
+
+BUFFER_SIZE = 1024 * 1024  # 1MB
```

Comment:
[Question] How did you decide this number?
It is just my opinion, but I think it is better to set this number as high as possible considering loop counts and the overhead we incur.

Reply:
That makes sense. Could you give us some recommendations?

Reply:
I don't have a strong reason here. 
How much do you think is a good value to set? How about 1GB?

Reply:
Instead of looping over manually, I decided to use `shutil.copyfileobj()` and let the default values of this function take care of this problem. Thank you for the discussion.